src/classes/OPP_StageMappingCMT.cls

```diff
@@ -0,0 +1,193 @@
+        }       
+        
+        // Take the value, if there is one, and create a finalized Opp Stage to State custom metadata type record. 
+        // Only create the pledged opp stage to state record if there's no existing opportunity stage to state records at all. 
+        if (overallCount == 0 && hasPledgedValue) {
+            metadataToInsert.add(createCustomMetadata(normalPledgeValue, COMMITTED_STATE_VAL));
+        }
+
+        return metadataToInsert;
+    }
+
+    /*******************************************************************************************************
+    * @description Creates a custom metadata record based on the parameters provided to the method. 
+    ********************************************************************************************************/
+    public static Metadata.CustomMetadata createCustomMetadata(String oppStage, String oppState) {        
+        Metadata.CustomMetadata customMd = new Metadata.CustomMetadata();
+        String mdTypeName = UTIL_Namespace.StrTokenNSPrefix(OPP_STAGE_TO_STATE_METADATA_NAME);
+
+        customMd.fullName =  mdTypeName + '.' + oppState + '_' + datetime.now().getTime();
+        customMd.label = oppState + '_' + datetime.now().getTime();
+        customMd.protected_x = false;
+
+        Map<String, Object> fieldMapping = new Map<String, Object>();
+        fieldMapping.put(OPP_STAGE_FIELD, oppStage);
+        fieldMapping.put(OPP_STATE_FIELD, oppState);
+
+        Metadata.CustomMetadata md = new Metadata.CustomMetadata();
+        md = CMT_MetadataAPI.addFieldsToMetadataObject(customMd, fieldMapping);
+
+        return md;
+    }
+
+    /*******************************************************************************************************
+    * @description Deploys the Custom metadata records using the OppStageToStateMetadataHandler as the handler
+    ********************************************************************************************************/
+    public static void deployOppStageToStateMetadata(List<Metadata.CustomMetadata> listOfMetadataToInsert, Map<String, Object> params) {
+        // Deploy the metadata 
+        String jobId = CMT_MetadataAPI.getUniqueJobId();
+        CMT_MetadataAPI.MetadataCallbackHandler oppCallbackHandler = new OppStageToStateMetadataHandler();
+        CMT_MetadataAPI.MetadataCallBack callback = new CMT_MetadataAPI.MetadataCallBack(jobId);
+        callback.callbackHandler = oppCallbackHandler;
+        callback.callbackParams = params;
+
+        CMT_MetadataAPI.deployMetadata(listOfMetadataToInsert, callback);
+    }
+
+    public class OppStageToStateMetadataHandler implements CMT_MetadataAPI.MetadataCallbackHandler {
+
+        public Map<String, Object> params;
+
+
+        /**************************************************************************************************************
+        * @description success handler - No implementation as of right now, will be implemented later as needed. 
```

Comment:
If these are not used is there any benefit in including them in the code base? It's extra code that needs to be tested, but it's not currently providing any value to the application. If there is no use case for it now and the comment says "as needed" I'm assuming there is no future plans to implement it either. If true, I would just remove the unused methods.

Reply:
I will defer to @KietTruongSFDO  on this, but I believe we are required to have a success handler for the callback, even if it does not do anything.  If not, I agree we can eliminate it.

Reply:
I was looking for that method name being used in the PR, but didn't see it referenced.

Reply:
It's required to be implemented. They're both methods of the interface "CMT_MetadataAPI.MetadataCallbackHandler"